src/components/pages/Login.tsx

```diff
@@ -37,6 +39,8 @@ export default function Login() {
     user: loggedInUser,
   } = useLoginUser()
 
+  const { login } = useContext(UserContext) as UserContextType
```

Comment:
does this work? `useContext<UserContextType>(UserContext)`

Reply:
that gives an error :
"Argument of type 'Context<UserContextType | null>' is not assignable to parameter of type 'Context<UserContextType>'.
  The types of 'Provider.propTypes' are incompatible between these types."

I think it's because of this line in the UserContext component ? :

```const UserContext = createContext<UserContextType | null>(null)```

Reply:
or just leave createContext<UserContextType> without null. and `const { login } = useContext(UserContext)`. it should be automatically typed when use `useContext(UserContext)` even without `as`

Reply:
I think that's what I tried initially but i got this error:
"Argument of type 'null' is not assignable to parameter of type 'UserContextType'."

```createContext<UserContextType>()``` will throw error that I need a default value which is what I set to be null. What is the best default value that can solve this?

Reply:
I will merge this one, and actually I changed a few things around userAuth in my next PR so I can make any changes regards to this on the next one as well. thanks

Reply:
```export type UserContextType = {
  currentUser: LoginPayload | null
  setUser: (loginData: LoginPayload) => void
  clearUser: () => void
}

const UserContext = createContext<UserContextType>({
  currentUser: null,
  setUser: () => undefined,
  clearUser: () => undefined,
})```

Setting Default value as this instead of 'null' seems to work, but let me know if this is not ideal